Update alm-examples in MongoDB CSV

BenElgar opened the pull request:
This PR is for internal review. Once approved I'll make a PR into the upstream.

This removes the status fields from the examples, addressing https://github.com/mongodb/mongodb-enterprise-kubernetes/issues/48, and updates them to use non-deprecated fields. I also made it a lot easier to read them.


For easy comparison, this is what the unescaped fields were previously:
```json
[
  {
    "kind": "MongoDB",
    "apiVersion": "mongodb.com/v1",
    "metadata": {
      "name": "sample-replica-set"
    },
    "spec": {
      "version": "4.0.10",
      "type": "ReplicaSet",
      "project": "my-project",
      "persistent": false,
      "members": 3,
      "credentials": "my-credentials"
    },
    "status": {
      "version": "4.0.10",
      "type": "ReplicaSet",
      "phase": "Running",
      "members": 3,
      "link": "https://cloud.mongodb.com/v2/5d35f7cdbf994268c3",
      "lastTransition": "2019-07-23T09:00:00Z"
    }
  },
  {
    "apiVersion": "mongodb.com/v1",
    "kind": "MongoDB",
    "metadata": {
      "name": "sample-sharded-cluster"
    },
    "spec": {
      "configServerCount": 3,
      "credentials": "my-credentials",
      "mongodsPerShardCount": 3,
      "mongosCount": 2,
      "persistent": false,
      "project": "my-project",
      "shardCount": 1,
      "type": "ShardedCluster",
      "version": "3.6.8"
    },
    "status": {
      "lastTransition": "2019-07-23T09:00:00Z",
      "phase": "Running"
    }
  },
  {
    "apiVersion": "mongodb.com/v1",
    "kind": "MongoDBUser",
    "metadata": {
      "name": "sample-user"
    },
    "spec": {
      "db": "$external",
      "project": "my-project",
      "roles": [
        {
          "db": "admin",
          "name": "clusterAdmin"
        }
      ],
      "username": "CN=mms-user-1,OU=cloud,O=MongoDB,L=New York,ST=New York,C=US"
    }
  },
  {
    "apiVersion": "mongodb.com/v1",
    "kind": "MongoDBOpsManager",
    "metadata": {
      "name": "ops-manager"
    },
    "spec": {
      "version": "4.2.0",
      "adminCredentials": "ops-manager-admin",
      "configuration": {
        "mms.fromEmailAddr": "admin@thecompany.com"
      },
      "applicationDatabase": {
        "members": 3,
        "version": "4.2.0",
        "persistent": true,
        "podSpec": {
          "cpu": "0.25"
        }
      }
    }
  }
]
```

BenElgar commented:
@chatton @rodrigovalin @LouisPlisso @antonlisovenko to review

antonlisovenko commented on upstream-community-operators/mongodb-enterprise/1.3.1/mongodb-enterprise.v1.3.1.clusterserviceversion.yaml:
```diff
@@ -1,36 +1,93 @@
+            },
+            "shardCount": 1,
+            "type": "ShardedCluster",
+            "version": "3.6.8"
```
should we use 4.2+ instead?

BenElgar replied:
Done

antonlisovenko reviewed (approved):
cool, I didn't know it's possible to use json there 👍

BenElgar committed:
Update MongoDB alm-examples

This removes the status fields from the examples, addressing
https://github.com/mongodb/mongodb-enterprise-kubernetes/issues/48, and
updates them to use non-deprecated fields. I also made it a lot easier
to read them.

Signed-off-by: Ben Elgar <me@benelgar.com>
